main/src/library/Random.flix

```diff
@@ -58,6 +58,26 @@ mod Random {
         let Random(o) = r;
         nextDouble(o)
 
+    ///
+    /// Returns the next pseudorandom 8-bit integer value from the given random number generator `r`.
+    ///
+    pub def nextInt8(r: Random): Int8 \ IO =
+        import static java.lang.Integer.valueOf(Int32): ##java.lang.Integer \ {};
+        import java.lang.Integer.byteValue(): Int8 \ {};
+        nextInRange(r, minInclusive = Int8.toInt32(Int8.minValue()), maxExclusive = Int8.toInt32(Int8.maxValue()) + 1)
+        |> valueOf
+        |> byteValue
+
+    ///
+    /// Returns the next pseudorandom 16-bit integer value from the given random number generator `r`.
+    ///
+    pub def nextInt16(r: Random): Int16 \ IO =
+        import static java.lang.Integer.valueOf(Int32): ##java.lang.Integer \ {};
+        import java.lang.Integer.shortValue(): Int16 \ {};
+        nextInRange(r, minInclusive = Int16.toInt32(Int16.minValue()), maxExclusive = Int16.toInt32(Int16.maxValue()) + 1)
+        |> valueOf
+        |> shortValue
```

Comment:
Any reason not to just create a random Int32 and then immediately convert it to an Int16? 

I think that would be simpler.

Reply:
Let me preface that I'm a random number generation noob.

> Any reason not to just create a random Int32 and then immediately convert it to an Int16?

I don't know if doing that would make the generated pattern no longer uniform. Do you know?